crypto/src/swap.rs

```diff
@@ -1,13 +1,43 @@
 use crate::ka;
 use crate::transaction::Fee;
 use anyhow::Result;
+use chacha20poly1305::{
+    aead::{Aead, NewAead},
+    ChaCha20Poly1305, Key, Nonce,
+};
+use decaf377::FieldExt;
+use once_cell::sync::Lazy;
 use penumbra_proto::{dex as pb, Protobuf};
 
 use crate::asset::Id as AssetId;
 use crate::keys::OutgoingViewingKey;
 
-// TODO: not sure what byte size is necessary here yet
-pub const SWAP_CIPHERTEXT_BYTES: usize = 128;
+// Swap ciphertext byte length
+pub const SWAP_CIPHERTEXT_BYTES: usize = 169;
+// Swap plaintext byte length
+pub const SWAP_LEN_BYTES: usize = 153;
+pub const OVK_WRAPPED_LEN_BYTES: usize = 80;
+
+/// The nonce used for swap encryption.
+///
+/// The nonce will always be `[0u8; 12]` which is okay since we use a new
+/// ephemeral key each time.
+pub static SWAP_ENCRYPTION_NONCE: Lazy<[u8; 12]> = Lazy::new(|| [0u8; 12]);
```

Comment:
Hmm, is this OK? If the ephemeral key used for symmetric crypto is the same one used for the `NotePayload`, I think we need to make sure we have distinct nonces for encrypting the note payload, the memo, the swap plaintext, etc.

Reply:
We could do something like (since all 0s is used for note encryption, 1u128 is used for memo encryption):

```rust
pub static SWAP_ENCRYPTION_NONCE: Lazy<[u8; 12]> = Lazy::new(|| { let nonce_bytes = 2u128.to_le_bytes(); nonce_bytes[0..12].try_into().expect("nonce fits in array") });
```

(Not for this PR but it's probably a good idea to put all our nonces together somewhere in the future)